flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -1040,8 +1041,9 @@ public void configure(ReadableConfig configuration, ClassLoader classLoader) {
         configuration
                 .getOptional(StateChangelogOptions.ENABLE_STATE_CHANGE_LOG)
                 .ifPresent(this::enableChangelogStateBackend);
-        Optional.ofNullable(loadStateBackend(configuration, classLoader))
-                .ifPresent(this::setStateBackend);
+        configuration
+                .getOptional(StateBackendOptions.STATE_BACKEND)
+                .ifPresent(conf -> setStateBackend(loadStateBackend(configuration, classLoader)));
```

Comment:
I'm not sure it is right to setStateBackend here if user does not explicitly config one. IIUC, the one set here will be the `fromApplication` in `StreamTask.createStateBackend`, which overrides the cluster configuration.

Reply:
@1996fanrui Thanks for your explanation! I missed the point of `getOptional`. Thus I have no more question. +1 from my side.